src/trackers/impressionsTracker.ts

```diff
@@ -80,7 +56,7 @@ export function impressionsTrackerFactory(
         for (let i = 0; i < impressionsCount; i++) {
```

Comment:
We are iterating the `impressionsToListener` array now. So we should use something like `const impressionsToListenerCount = impressionsToListener.length;`, in case some strategy changes the array.